cmd/livepeer_bench/livepeer_bench.go

```diff
@@ -136,13 +138,23 @@ func main() {
 				t := time.Now()
 				_, err := tc.Transcode(in, out)
 				end := time.Now()
-				fmt.Printf("%s,%d,%d,%0.4v\n", end.Format("2006-01-02 15:04:05.9999"), k, j, end.Sub(t).Seconds())
 				if err != nil {
 					glog.Fatalf("Transcoding failed for session %d segment %d: %v", k, j, err)
 				}
-				if k == 0 {
-					segTotalDur += v.Duration
-					segCount++
+				fmt.Printf("%s,%d,%d,%0.4v,%0.4v\n", end.Format("2006-01-02 15:04:05.9999"), k, j, v.Duration, end.Sub(t).Seconds())
+				segTxDur := end.Sub(t).Seconds()
+				mu.Lock()
+				transcodeDur += segTxDur
+				srcDur += v.Duration
+				segCount++
+				if segTxDur <= v.Duration {
+					realTimeSegCount += 1
+				}
+				mu.Unlock()
+				iterEnd := time.Now()
+				segDur := time.Duration(int64(v.Duration*1000)) * time.Millisecond
```

Comment:
This can be shortened to:

```golang
segDur := time.Duration(v.Duration * float64(time.Second))
```

`time.Second` is stored in milliseconds so multiplying it by `v.Duration`, which is stored in seconds, will give us `v.Duration` in terms of milliseconds which can then be converted into a `time.Duration`.

Reply:
Neat thx!